### PR TITLE
Added webhook (tx-confirmation) support

### PR DESCRIPTION
**Pull request by vtnerd**
This is a "still-in-progress" PR/patch to add webhook support to `monero-lws-daemon`. I would like to add some unit testing before officially merging this PR. There's also some trivial stuff I noticed - wrong copyright dates for instance. This branch should be considered "beta" - the expectation is that is should be near usable in a production environment.

This PR currently supports "`tx-confirmation`" only - a Monero address + (optional) payment_id is sent via REST, along with a provided URL. The URL is then invoked 1+ times when an output for that address+payment_id pair is found on the blockchain. The number of URL invokes is determine in the original API call - the user specifies the number of confirmations needed (defaults to 1).

Please respond to this PR if you need help using this new feature. I **will** perform force pushes on this branch to keep the history lean and mean so be warned about that.

A quick example (which is from the documentation):

#### Initial user invoke
```json
{
  "auth": "f50922f5fcd186eaa4bd7070b8072b66fea4fd736f06bd82df702e2314187d09",
  "params": {
    "type": "tx-confirmation",
    "url": "http://127.0.0.1:7000",
    "payment_id": "df034c176eca3296",
    "token": "1234",
    "address": "9uTcr6T9GURRt7UADQc2rhjg5oMYBDyoQ5jgx8nAvVvs757WwDkc2vHLPJhwZfCnfVdnWNvuuKzJe8eMVTKwadYzBrYRG5j"
  }
}
```

### Initial Response
```json
{
  "payment_id": "df034c176eca3296",
  "event_id": "fa10a4db485145f1a24dc09c19a79d43",
  "token": "1234",
  "confirmations": 1,
  "url": "http://127.0.0.1:7000"
}
```

### Triggered Response (to specified URL)
```json
{
  "event": "tx-confirmation",
  "payment_id": "df034c176eca3296",
  "token": "1234",
  "confirmations": 1,
  "id": "fa10a4db485145f1a24dc09c19a79d43",
  "tx_info": {
    "id": {
      "high": 0,
      "low": 5550229
    },
    "block": 2192100,
    "index": 0,
    "amount": 4949570000,
    "timestamp": 1678324181,
    "tx_hash": "901f9a2a919b6312131537ff6117d56ce2c0dc1f1341b845d7667299e1ef892f",
    "tx_prefix_hash": "89685cb7acb836fde30fae8be5d8b884e92706df086960d0508e146979ef80dc",
    "tx_public": "54c153792e47c1da8ceb3979560c424c1928b7b4a089c1c8b3ce99c563e1d240",
    "rct_mask": "f3449407dc3721299b5309c0c336a17daeebce55165ddd447ba28bbd1f46c201",
    "payment_id": "df034c176eca3296",
    "unlock_time": 0,
    "mixin_count": 15,
    "coinbase": false
  }
}
```

### CODE REVIEW

File: docs/administration.md [hunk 1]
```diff
@@ -45,34 +45,283 @@ to put the account into the "inactive" state. Deleting accounts is not
...
+will also fail if `https` verification is mismatched. The rule is: (1) if
+the callback server has SSL/TLS disabled, the webhook should use `http://`,
+(2) if the callback server has a self-signed certificate, `https://` and
+`--webhook-ssl-verification none` should be used, and (3) if the callback
+server is using "Let's Encrypt" (or similar), then `https://` with no
+additional command line flag should be used.
+
+
+#### Initial Request to server
+Example where admin authentication is required (`--disable-admin-auth` NOT set on start):
```
Comment (serhack) [inline docs/administration.md:169]: Could you rephrase that to include the default behavior of monero-lws? :)

**Review by serhack — APPROVED** [review]
As always, great work :)

**Comment by CryptoGrampy** [comment]
Very cool.  I'm assuming at least a couple people will be using these for payment validation, so some food for thought (maybe add a signature that can be validated?):

https://stripe.com/docs/webhooks/signatures
https://docs.github.com/en/webhooks-and-events/webhooks/securing-your-webhooks

**Comment by vtnerd** [comment]
@CryptoGrampy that's likely going to be a in a future change, if at all. The secret token provides what most users need.

**Comment by vtnerd** [comment]
I should also update, I'm not against the signature, just against adding it right now as I've got many other things of higher priority.

> I'm assuming at least a couple people will be using these for payment validation

There is at least one business attempting that right now (they are still trying to deploy).

File: docs/administration.md (hunk 1 above)
Comment (vtnerd) [reply docs/administration.md:169]: Done.
